### ISSUE TITLE
Batali fails to detect dependencies when they are added in a loop

### ISSUE THREAD

**Issue by cwjohnston** [hw-labs/batali#26]
e.g. of the dependencies defined for the 'alice' cookbook [here](https://github.com/cwjohnston/chef-alice/blob/6f8e8a96db520fba7bc44ac3ae6630f65c24c5a5/metadata.rb#L8-L10), the manifest lists only "runit":

Batali file from cookbook which wraps 'alice':
```
Batali.define do

  source 'https://supermarket.chef.io'
  metadata

  cookbook 'alice', git: 'https://github.com/cwjohnston/chef-alice.git', :ref => 'master'

  # integration test dependencies
  cookbook 'apt'
  cookbook 'avahi', git: 'https://github.com/needle-cookbooks/chef-avahi.git', ref: 'v0.1.0'
  cookbook 'hardchats-test', path: './test/cookbooks/hardchats-test'

end
```

batali.manifest of same
```
{
  "infrastructure": false,
  "cookbook": [
    {
      "name": "alice",
      "dependencies": [
        [
          "runit",
          "~> 1.1.6"
        ]
      ],
      "version": "0.3.3",
      "source": {
        "url": "https://github.com/cwjohnston/chef-alice.git",
        "ref": "6f8e8a96db520fba7bc44ac3ae6630f65c24c5a5",
        "type": "Batali::Source::Git"
      }
    },
    {
      "name": "runit",
      "dependencies": [
        [
          "build-essential",
          ">= 0.0.0"
        ],
        [
          "yum",
          ">= 0.0.0"
        ]
      ],
      "version": "1.1.6",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/runit/versions/1.1.6/download",
        "version": "1.1.6"
      }
    },
    {
      "name": "build-essential",
      "dependencies": [

      ],
      "version": "2.2.3",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/build-essential/versions/2.2.3/download",
        "version": "2.2.3"
      }
    },
    {
      "name": "yum",
      "dependencies": [

      ],
      "version": "3.6.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/yum/versions/3.6.0/download",
        "version": "3.6.0"
      }
    },
    {
      "name": "apt",
      "dependencies": [

      ],
      "version": "2.7.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/apt/versions/2.7.0/download",
        "version": "2.7.0"
      }
    },
    {
      "name": "avahi",
      "dependencies": [
        [
          "git",
          "> 0"
        ]
      ],
      "version": "0.1.0",
      "source": {
        "url": "https://github.com/needle-cookbooks/chef-avahi.git",
        "ref": "25e269c6a6b595a6c00aa2525e2fd45067b262ef",
        "type": "Batali::Source::Git"
      }
    },
    {
      "name": "git",
      "dependencies": [
        [
          "build-essential",
          ">= 0.0.0"
        ],
        [
          "dmg",
          ">= 0.0.0"
        ],
        [
          "windows",
          ">= 0.0.0"
        ],
        [
          "yum-epel",
          ">= 0.0.0"
        ]
      ],
      "version": "4.2.2",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/git/versions/4.2.2/download",
        "version": "4.2.2"
      }
    },
    {
      "name": "dmg",
      "dependencies": [

      ],
      "version": "2.2.2",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/dmg/versions/2.2.2/download",
        "version": "2.2.2"
      }
    },
    {
      "name": "windows",
      "dependencies": [
        [
          "chef_handler",
          ">= 0.0.0"
        ]
      ],
      "version": "1.36.6",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/windows/versions/1.36.6/download",
        "version": "1.36.6"
      }
    },
    {
      "name": "chef_handler",
      "dependencies": [

      ],
      "version": "1.1.6",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/chef_handler/versions/1.1.6/download",
        "version": "1.1.6"
      }
    },
    {
      "name": "yum-epel",
      "dependencies": [
        [
          "yum",
          "~> 3.0"
        ]
      ],
      "version": "0.6.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/yum-epel/versions/0.6.0/download",
        "version": "0.6.0"
      }
    },
    {
      "name": "hardchats-test",
      "dependencies": [
        [
          "avahi",
          "> 0"
        ],
        [
          "database",
          "> 0"
        ],
        [
          "hardchats",
          "> 0"
        ]
      ],
      "version": "0.1.0",
      "source": {
        "type": "Batali::Source::Path",
        "path": "./test/cookbooks/hardchats-test"
      }
    },
    {
      "name": "database",
      "dependencies": [
        [
          "postgresql",
          ">= 1.0.0"
        ]
      ],
      "version": "4.0.5",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/database/versions/4.0.5/download",
        "version": "4.0.5"
      }
    },
    {
      "name": "postgresql",
      "dependencies": [
        [
          "apt",
          ">= 1.9.0"
        ],
        [
          "build-essential",
          ">= 0.0.0"
        ],
        [
          "openssl",
          "~> 4.0.0"
        ]
      ],
      "version": "3.4.18",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/postgresql/versions/3.4.18/download",
        "version": "3.4.18"
      }
    },
    {
      "name": "openssl",
      "dependencies": [
        [
          "chef-sugar",
          ">= 0.0.0"
        ]
      ],
      "version": "4.0.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/openssl/versions/4.0.0/download",
        "version": "4.0.0"
      }
    },
    {
      "name": "chef-sugar",
      "dependencies": [

      ],
      "version": "3.1.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/chef-sugar/versions/3.1.0/download",
        "version": "3.1.0"
      }
    },
    {
      "name": "hardchats",
      "dependencies": [
        [
          "certificate",
          "> 0"
        ],
        [
          "database",
          "> 0"
        ],
        [
          "mysql",
          "> 0"
        ],
        [
          "logrotate",
          "> 0"
        ],
        [
          "alice",
          "~> 0.3.1"
        ],
        [
          "nginx",
          "~> 1.7.0"
        ],
        [
          "swap",
          "~> 0.3.5"
        ],
        [
          "backup",
          "> 0"
        ],
        [
          "build-essential",
          "> 0"
        ]
      ],
      "version": "0.2.2",
      "source": {
        "type": "Batali::Source::Path",
        "path": "/Users/cwj/hardchats/chef-repo/site-cookbooks/hardchats"
      }
    },
    {
      "name": "certificate",
      "dependencies": [
        [
          "chef-vault",
          ">= 0.0.0"
        ]
      ],
      "version": "1.0.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/certificate/versions/1.0.0/download",
        "version": "1.0.0"
      }
    },
    {
      "name": "chef-vault",
      "dependencies": [

      ],
      "version": "1.3.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/chef-vault/versions/1.3.0/download",
        "version": "1.3.0"
      }
    },
    {
      "name": "mysql",
      "dependencies": [
        [
          "yum-mysql-community",
          ">= 0.0.0"
        ],
        [
          "smf",
          ">= 0.0.0"
        ]
      ],
      "version": "6.0.21",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/mysql/versions/6.0.21/download",
        "version": "6.0.21"
      }
    },
    {
      "name": "yum-mysql-community",
      "dependencies": [
        [
          "yum",
          ">= 3.0"
        ]
      ],
      "version": "0.1.17",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/yum-mysql-community/versions/0.1.17/download",
        "version": "0.1.17"
      }
    },
    {
      "name": "smf",
      "dependencies": [
        [
          "rbac",
          ">= 1.0.1"
        ]
      ],
      "version": "2.2.6",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/smf/versions/2.2.6/download",
        "version": "2.2.6"
      }
    },
    {
      "name": "rbac",
      "dependencies": [

      ],
      "version": "1.0.3",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/rbac/versions/1.0.3/download",
        "version": "1.0.3"
      }
    },
    {
      "name": "logrotate",
      "dependencies": [

      ],
      "version": "1.9.1",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/logrotate/versions/1.9.1/download",
        "version": "1.9.1"
      }
    },
    {
      "name": "nginx",
      "dependencies": [
        [
          "build-essential",
          ">= 0.0.0"
        ],
        [
          "yum",
          ">= 0.0.0"
        ],
        [
          "apt",
          ">= 0.0.0"
        ],
        [
          "runit",
          ">= 0.0.0"
        ],
        [
          "ohai",
          ">= 1.1.4"
        ]
      ],
      "version": "1.7.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/nginx/versions/1.7.0/download",
        "version": "1.7.0"
      }
    },
    {
      "name": "ohai",
      "dependencies": [

      ],
      "version": "2.0.1",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/ohai/versions/2.0.1/download",
        "version": "2.0.1"
      }
    },
    {
      "name": "swap",
      "dependencies": [

      ],
      "version": "0.3.8",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/swap/versions/0.3.8/download",
        "version": "0.3.8"
      }
    },
    {
      "name": "backup",
      "dependencies": [
        [
          "cron",
          ">= 0.0.0"
        ],
        [
          "gem_specific_install",
          ">= 0.0.0"
        ]
      ],
      "version": "1.3.0",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/backup/versions/1.3.0/download",
        "version": "1.3.0"
      }
    },
    {
      "name": "cron",
      "dependencies": [

      ],
      "version": "1.6.1",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/cron/versions/1.6.1/download",
        "version": "1.6.1"
      }
    },
    {
      "name": "gem_specific_install",
      "dependencies": [
        [
          "git",
          ">= 0.0.0"
        ]
      ],
      "version": "0.2.4",
      "source": {
        "type": "Batali::Source::Site",
        "url": "https://supermarket.chef.io/api/v1/cookbooks/gem_specific_install/versions/0.2.4/download",
        "version": "0.2.4"
      }
    }
  ]
}
```

**Comment by chrisroberts** [comment]
Nice find. I was being lazy and only partially parsing the files, which led to missing dynamically defined `depends`. Now performing full file parse on `metadata.rb` files which fixes this issue.

**Comment by chrisroberts** [comment]
This fix was released 0.2.10
